packages/stream-metadata/src/evmRpcClient.ts

```diff
@@ -1,44 +1,24 @@
-import { config } from './environment'
-import { createPublicClient, http } from 'viem'
-import { riverChainDevnet, riverChainLocalhost, riverChainProduction } from './evmChainConfig'
+import { RiverRegistry } from '@river-build/web3'
+import { ethers } from 'ethers'
 
-const publicClient: Record<number, ReturnType<typeof createPublicClientFromChainId>> = {}
+import { Config } from './types'
 
-function createPublicClientFromChainId(chainId: number) {
-	let riverChainUrl: string
+let riverRegistry: ReturnType<typeof createRiverRegistry> | undefined
 
-	switch (chainId) {
-		case 550:
-			riverChainUrl = riverChainProduction.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainProduction,
-				transport: http(riverChainUrl),
-			})
-		case 6524490:
-			riverChainUrl = riverChainDevnet.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainDevnet,
-				transport: http(riverChainUrl),
-			})
-		case 31338:
-			riverChainUrl = riverChainLocalhost.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainLocalhost,
-				transport: http(riverChainUrl),
-			})
-		default:
-			console.error(`Unsupported chain ID: ${chainId}`)
-			throw new Error(`Unsupported chain ${chainId}`)
+function createRiverRegistry(config: Config) {
```

Comment:
This function exists in web3 https://github.com/river-build/river/blob/main/packages/web3/src/RiverRegistryFactory.ts#L5